### PR TITLE
LeastRequestedPriority/MostRequestedPriority/BalancedResourceAllocation as Score plugins

### PR DESCRIPTION
**Pull request by zouyee**
Signed-off-by: Zou Nengren <zouyee1989@gmail.com>

/assign @ahg-g

**What type of PR is this?**
/kind feature
/priority important-soon

**What this PR does / why we need it**:
 MostRequestedPriority as Score plugin

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
 MostRequestedPriority as Score plugin
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/mostrequestedpriority/most_requested.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+// MostRequestedPriority is a priority plugin that favors nodes with most requested resources.
+type MostRequestedPriority struct {
+	handle framework.FrameworkHandle
+}
+
+var _ = framework.ScorePlugin(&MostRequestedPriority{})
+
+// Name is the name of the plugin used in the plugin registry and configurations.
+const Name = "MostRequestedPriority"
```
Comment (hex108) [inline pkg/scheduler/framework/plugins/mostrequestedpriority/most_requested.go]: If we follow the pattern as `imagelocality` and `nodeaffinity`, we could name it to `MostRequested`. Then other places also do not need the `priority` suffix.

**Review by hex108 — COMMENTED** [review]
Thanks @zouyee ! Some nits.

**Comment by zouyee** [comment]
/remove-area apiserver
/remove-area cloudprovider
/remove-area dependency
/remove-area e2e-test-framework
/remove-area ipvs
/remove-area kubeadm
/remove-area kubectl kubelet test
/remove-kind feature
/remove-sig api-machinery apps auth cli cloud-provider cluster-lifecycle network storage testing

File: pkg/scheduler/framework/plugins/mostrequestedpriority/most_requested.go (hunk 1 above)
Comment (zouyee) [reply pkg/scheduler/framework/plugins/mostrequestedpriority/most_requested.go]: done! PTAL

File: pkg/scheduler/framework/plugins/mostrequested/most_requested.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/scheduler/algorithm/priorities"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/migration"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+)
+
+// MostRequested is a priority plugin that favors nodes with most requested resources.
```
Comment (hex108) [inline pkg/scheduler/framework/plugins/mostrequested/most_requested.go]: s/MostRequested is a priority plugin/MostRequested is a plugin/

File: pkg/scheduler/framework/plugins/mostrequested/most_requested.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (hex108) [inline pkg/scheduler/framework/plugins/mostrequested/most_requested.go]: s/2016/2019/

**Review by hex108 — COMMENTED** [review]
Others LGTM.

**Comment by hex108** [comment]
Thanks!

/lgtm

**Comment by zouyee** [comment]
/remove-area apiserver
/remove-area cloudprovider
/remove-area dependency
/remove-area e2e-test-framework
/remove-area ipvs
/remove-area kubeadm
/remove-area kubectl kubelet test
/remove-kind feature
/remove-sig api-machinery apps auth cli cloud-provider cluster-lifecycle network storage testing

**Pull request closed without merging**